File: java_MVC/dao/daoImpl/PushImpl.java
package dao.daoImpl;

import dao.Push;
import entity.Filed;

public class PushImpl implements Push {

    @Override
    public void push(Filed stack, int data) {
        if (stack.getSp() < stack.getStack().length) {
            stack.getStack()[stack.getSp()] = data;
            stack.setSp(stack.getSp() + 1);
        } else {
            System.out.println("スタック領域に空きがない");
        }
    }
}

File: python_MVC/dao/daoImpl/PushImpl.py
# daoImpl/pushImpl.py

from dao.Push import Push
from entity.Filed import Filed

class PushImpl(Push):
    def push(self, stack: Filed, data: int):
        if stack.getSp() < len(stack.getStack()):
            stack.getStack()[stack.getSp()] = data
            stack.setSp(stack.getSp() + 1)
        else:
            print("スタック領域に空きがない")
